packages/block-editor/src/components/provider/index.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { DropZoneProvider, SlotFillProvider } from '@wordpress/components';
+import { withDispatch, RegistryConsumer } from '@wordpress/data';
+import { createHigherOrderComponent, compose } from '@wordpress/compose';
+
+/**
+ * Higher-order component which renders the original component with the current
+ * registry context passed as its `registry` prop.
+ *
+ * @param {WPComponent} OriginalComponent Original component.
+ *
+ * @return {WPComponent} Enhanced component.
+ */
+const withRegistry = createHigherOrderComponent(
```

Comment:
Aside: It seems a common enough pattern that we should consider one or both of opting for [`useContext` hook](https://reactjs.org/docs/hooks-reference.html#usecontext) from React 16.8.0, and/or our own implementation of a `withContext` higher-order component which would behave similarly.

The main difference with a higher-order component is that we'd need to instruct the prop name to which the context value should be assigned. One or both of:

```js
withContext( 'registry', RegistryContext )( MyComponent );
withContext( { registry: RegistryContext } )( MyComponent );
```

Reply:
Maybe we should try hooks for a moment and see whether adding higher order components still make sense or not.